sklearnex/tests/test_monkeypatch.py

```diff
@@ -27,142 +27,166 @@ def test_monkey_patching():
         for c in v:
             _classes.append(c[0])
 
-    sklearnex.patch_sklearn()
-
-    for i, _ in enumerate(_tokens):
-        t = _tokens[i]
-        p = _classes[i][0]
-        n = _classes[i][1]
-
-        class_module = getattr(p, n).__module__
-        assert class_module.startswith("daal4py") or class_module.startswith(
-            "sklearnex"
-        ), "Patching has completed with error."
-
-    for i, _ in enumerate(_tokens):
-        t = _tokens[i]
-        p = _classes[i][0]
-        n = _classes[i][1]
-
-        sklearnex.unpatch_sklearn(t)
-        class_module = getattr(p, n).__module__
-        assert class_module.startswith("sklearn"), "Unpatching has completed with error."
-
-    sklearnex.unpatch_sklearn()
-
-    for i, _ in enumerate(_tokens):
-        t = _tokens[i]
-        p = _classes[i][0]
-        n = _classes[i][1]
-
-        class_module = getattr(p, n).__module__
-        assert class_module.startswith("sklearn"), "Unpatching has completed with error."
-
-    sklearnex.unpatch_sklearn()
-
-    for i, _ in enumerate(_tokens):
-        t = _tokens[i]
-        p = _classes[i][0]
-        n = _classes[i][1]
-
-        sklearnex.patch_sklearn(t)
-
-        class_module = getattr(p, n).__module__
-        assert class_module.startswith("daal4py") or class_module.startswith(
-            "sklearnex"
-        ), "Patching has completed with error."
-
-    sklearnex.unpatch_sklearn()
+    try:

```

Comment:
is it good to add it to test? will we catch any exceptions and errors this way?

Reply:
https://docs.python.org/3/reference/compound_stmts.html#finally

Reply:
It essentially insured that the sklearn patch will be reverted even if an exception/error is raised